From 173b3a6268dd056b16076484096c4f73506e83b4 Mon Sep 17 00:00:00 2001
From: Rajeev Kulkarni <rajeevk@ti.com>
Date: Mon, 10 Jan 2011 01:17:18 -0600
Subject: OMAP4: Proper Mutex Handling to avoid Race condition in SR

DVFS and OPP changing involves SR enable and disable.
Because of Improper Mutex handling system was crashing
at SR Enable.

Change-Id: I04c80eb94809f7315f0c3410e0dfdc86244c6214
Signed-off-by: Rajeev Kulkarni <rajeevk@ti.com>
---
 arch/arm/mach-omap2/voltage.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

Index: Kernel/arch/arm/mach-omap2/voltage.c
===================================================================
--- Kernel.orig/arch/arm/mach-omap2/voltage.c	2011-01-19 00:10:05.072071045 +0530
+++ Kernel/arch/arm/mach-omap2/voltage.c	2011-01-19 00:19:16.763322079 +0530
@@ -2313,11 +2313,11 @@
 	if (!is_volt_scaled)
 		omap_voltage_scale_vdd(voltdm, volt);
 
-	mutex_unlock(&vdd->scaling_mutex);
-
 	/* Enable Smartreflex module */
 	omap_smartreflex_enable(voltdm);
 
+	mutex_unlock(&vdd->scaling_mutex);
+
 	/* Scale dependent vdds */
 	scale_dep_vdd(vdd);
 
